### DISCUSSION TITLE
Can you persist s3 credentials?

### DISCUSSION THREAD

**Discussion by riazarbi** [duckdb/duckdb#7161] [Q&A] [answered]
Hello, 

I am using duckdb v 0.7.1.

When using duckdb with a persistent database, I find that my s3 credentials do not persist across sessions. I can set credentials and run a query against s3 in a single session, but if I exit that session, and then start a new one, the settings are not there. 

Is it possible to have a particular duckdb database remember s3 credentials?

**Comment by Mause** [comment, marked as answer]
You can't persist any configuration items within the database itself at the moment.

You can however:

 * add lines to a .duckdbrc file: https://duckdb.org/docs/api/cli#configuring-the-cli
 * or set environment variables on your system: https://github.com/duckdb/duckdb/blob/b475a57930f0a6c5163c82186e74b18391250ab0/extension/httpfs/s3fs.cpp#L157-L163
